cmd/earthly/root_cmds.go

```diff
@@ -488,32 +500,76 @@ func symlinkEarthlyToEarth() error {
 	return nil
 }
 
-func (app *earthlyApp) actionDocker(cliCtx *cli.Context) error {
-	app.commandName = "docker"
-
-	dir := filepath.Dir(app.dockerfilePath)
-	earthfilePath := filepath.Join(dir, "Earthfile")
+func handleDockerFiles(buildContextPath string) error {
+	earthfilePath := filepath.Join(buildContextPath, "Earthfile")
 	earthfilePathExists, err := fileutil.FileExists(earthfilePath)
 	if err != nil {
 		return errors.Wrapf(err, "failed to check if %s exists", earthfilePath)
 	}
 	if earthfilePathExists {
 		return errors.Errorf("earthfile already exists; please delete it if you wish to continue")
 	}
-	defer os.Remove(earthfilePath)
 
-	err = docker2earthly.Docker2Earthly(app.dockerfilePath, earthfilePath, app.earthfileFinalImage)
+	dockerIgnorePath := filepath.Join(buildContextPath, ".dockerignore")
+	dockerIgnorePathExists, err := fileutil.FileExists(dockerIgnorePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to check if %s exists", dockerIgnorePath)
+	}
+
+	if !dockerIgnorePathExists {
+		return nil
+	}
+
+	earthlyIgnoreFilePath := filepath.Join(buildContextPath, ".earthlyignore")
+	err = copy.Copy(dockerIgnorePath, earthlyIgnoreFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to copy %s to %s", dockerIgnorePath, earthlyIgnoreFilePath)
+	}
+	defer os.Remove(earthlyIgnoreFilePath)
+	return nil
+}
+
+func (app *earthlyApp) actionDocker(cliCtx *cli.Context) error {
+	app.commandName = "docker"
+
+	flagArgs, nonFlagArgs, err := variables.ParseFlagArgsWithNonFlags(cliCtx.Args().Slice())
+	if err != nil {
+		return errors.Wrapf(err, "parse args %s", strings.Join(cliCtx.Args().Slice(), " "))
+	}
+	if len(nonFlagArgs) == 0 {
+		_ = cli.ShowAppHelp(cliCtx)
+		return errors.Errorf(
+			"no build context path provided. Try %s docker <path>", cliCtx.App.Name)
+	} else if len(nonFlagArgs) != 1 {
```

Comment:
Yeah, no worries, I'm the same way :) Not with `else` statements, but with other silly easy-to-miss stylistic mistakes - like bad documentation, for instance. I never catch that when I'm writing code, only when I'm reviewing it.

Long-winded style opinion: The main reason you most likely won't catch unnecessary `else` statements in my PRs is that I haven't used _any_ `else` statements in a very long time. Another engineer once told me to think of an early return as "popping information off of a developer's mental stack," and that made me realize that logic that merges back into a parent scope (like an `if` that doesn't end in a `return`) is pretty hard to read - because I'm forcing the reader to figure out which variables in the nested scope escape back to the parent scope, and mental merges of variable state are hard.

Around that time, I decided to _try_ just never using `else` statements, see how long it was before I started to miss them. It's been about five years now, and I'm still following that rule. In fact, I've started to lean toward an "every logic branch must end in a return" type of rule. More often than not, it just forces me to keep functions smaller, break logic out into helper functions more frequently, and limit how deeply I nest my scope. None of those are really bad things (most of the  time).